### PR TITLE
Integrated receipt_chain_database into Coda

### LINKED ISSUE #1116: Hook Receipt Chain database into Coda
- Make a simple instantiation of the receipt chain database and should be member variable of Coda main. Use `Core.Hashtbl`as instantiation. 
- Every time you make a transaction, via `schedule_payment` in `coda_main.ml`, add that transaction into the database

### PR DESCRIPTION
**Pull request by johnwu93**
Added `receipt_chain_database` into coda

I will make an integration test that shows that whenever a client sends a transaction, we can prove that the sender made that transaction the `receipt_chain_database`

Checklist:

- [X] Tests were added for the new behavior
- [X] All tests pass (CI will check this if you didn't)
- [X] Does this close issues? List them:

Closes #1116

### CODE REVIEW

File: src/app/cli/src/coda_main.ml [hunk 1]
```diff
@@ -989,33 +996,73 @@ module Run (Config_in : Config_intf) (Program : Main_intf) = struct
...
+
+  let schedule_payment log t (txn : Payment.t) =
+    let public_key = Public_key.compress txn.sender in
+    let account_opt = get_account t public_key in
+    if not (is_valid_payment t txn account_opt) then (
       Core.Printf.eprintf "Invalid payment: account balance is too low" ;
       Core.exit 1 ) ;
+    let receipt = record_payment ~log t txn (Option.value_exn account_opt) in
     let txn_pool = transaction_pool t in
     don't_wait_for (Transaction_pool.add txn_pool txn) ;
```
[the remark this answers is not in the document]
Comment (imeckler) [reply src/app/cli/src/coda_main.ml]: we should still record it though, no? this seems correct to me

[the remark this answers is not in the document]
Comment (imeckler) [reply src/app/cli/src/coda_main.ml]: even if it doesn't ultimately make it in we are keeping the whole tree of everything we ever send so it should be fine

[the remark this answers is not in the document]
Comment (johnwu93) [reply src/app/cli/src/coda_main.ml]: I agree with @imeckler, we should store it. We can see if the transaction made it by using the `Receipt_chain_database.prove`.

File: src/app/cli/src/coda_main.ml [hunk 2]
```diff
@@ -989,33 +996,73 @@ module Run (Config_in : Config_intf) (Program : Main_intf) = struct
...
+  let schedule_payments log t txns =
+    List.iter txns ~f:(fun txn ->
+        let _ : Receipt.Chain_hash.t = schedule_payment log t txn in
+        () )
+
+  let prove_receipt t ~proving_receipt ~resulting_receipt :
+      (Receipt.Chain_hash.t * Payment.t) list Deferred.t =
+    let receipt_chain_database = receipt_chain_database t in
+    let result =
+      Receipt_chain_database.prove receipt_chain_database ~proving_receipt
```
[the remark this answers is not in the document]
Comment (johnwu93) [reply src/app/cli/src/coda_main.ml]: Working on this issue will resolve this issue

https://github.com/CodaProtocol/coda/issues/1144

File: src/app/cli/src/coda_main.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnwu93) [reply src/app/cli/src/coda_main.ml]: I wrote an issue about this. We will solve this issue in a later PR 
 https://github.com/CodaProtocol/coda/issues/1143

**Pull request merged**